gwctl/pkg/resourcediscovery/discoverer.go

```diff
@@ -286,6 +287,11 @@ func fetchGatewayClasses(ctx context.Context, k8sClients *common.K8sClients, fil
 			return []gatewayv1.GatewayClass{}, err
 		}
 
+		// because api-server doesn't return TypeMeta in `gatewayClass`
+		gcApplyConfig := apisv1beta1.GatewayClass(gatewayClass.Name)
```

Comment:
Ah nice catch! This may just be a bit more involved. 

The thing is, because we construct the gatewayClass object with the v1 versioned struct (i.e.  `&gatewayv1.GatewayClass{}`, client-go will fetch the GatewayClass for the v1 version. This means the correct APIVersion would correspond to the `v1` package (and not apisv1beta1 = v1beta1):

```
gcApplyConfig := apisv1.GatewayClass(gatewayClass.Name)
```

Having said that, if we think about it a bit more, use of ApplyConfig isn't giving any real advantage here. How about something like this:

```
		gatewayClass.APIVersion = gatewayv1.GroupVersion.String()
		gatewayClass.Kind = reflect.TypeOf(gatewayClass).Elem().Name()
```

(Infact, using gatewayClass.Kind = "GatewayClass" would also just have been good enough)

Reply:
Ahh good point, will take care of this.